app/src/main/java/openfoodfacts/github/scrachx/openfood/features/allergensalert/AllergensAlertFragment.kt

```diff
@@ -131,46 +130,51 @@ class AllergensAlertFragment : NavigationBaseFragment() {
             productRepository.getAllergensByEnabledAndLanguageCode(false, LocaleHelper.getLanguage(requireActivity()))
                     .subscribeOn(Schedulers.io())
                     .observeOn(AndroidSchedulers.mainThread())
-                    .doOnError { it.printStackTrace() }
                     .map { allergens -> allergens.sortedBy { it.name } }
-                    .subscribe { allergens ->
-                        MaterialDialog.Builder(requireContext()).apply {
-                            title(R.string.title_dialog_alert)
-                            items(allergens.map { it.name })
-                            itemsCallback { _, _, position, _ ->
-                                productRepository.setAllergenEnabled(allergens[position].allergenTag, true)
-                                mAllergensEnabled!!.add(allergens[position])
-                                adapter.notifyItemInserted(mAllergensEnabled!!.size - 1)
-                                binding.allergensRecycle.scrollToPosition(adapter.itemCount - 1)
-                                matomoAnalytics.trackEvent(AnalyticsEvent.AllergenAlertCreated(allergens[position].allergenTag))
+                    .subscribe(
```

Comment:
I prefer to use `doOnError` so that we can then use the single lambda on subscribe. Is there any reason to put it like this?

Reply:
actually no reason, will revert to previous approach